src/App.test.js

```diff
@@ -0,0 +1,228 @@
+import React from "react";
+import { shallow } from "enzyme";
+import App from "./App";
+import { findByTestAttr, storeFactory } from "../test/testUtils";
+import { DataTable, Paginator, SearchField } from "lucid-ui";
+
+const setup = (initialState = {}) => {
+  const store = storeFactory(initialState);
+  const wrapper = shallow(<App store={store} />)
+    .dive()
+    .dive();
+  return wrapper;
+};
+
+describe("App component", () => {
+  test("should render component without error", () => {
+    const component = findByTestAttr(setup(), "App");
+    expect(component).toHaveLength(1);
+  });
+
+  describe("condintional rendering for fetch requests", () => {
+    test("should render `Something Went Wrong!` message in case of api fetch request failure", () => {
+      const component = findByTestAttr(
+        setup({ reducer: { error: true } }),
+        "Error"
+      );
+      expect(component.text()).toBe("Something Went Wrong!");
+    });
+
+    test("should not render `Something Went Wrong!` message in case of api fetch request failure", () => {
+      const component = findByTestAttr(
+        setup({ reducer: { error: false, todos_list: [] } }),
+        "Error"
+      );
+      expect(component).toHaveLength(0);
+    });
+  });
+
+  describe("condintional rendering todo_list", () => {
+    test("should render DataTable when todo_list array contains data", () => {
+      const component = findByTestAttr(
+        setup({
+          reducer: {
+            error: false,
+            todos_list: [
```

Comment:
don't repeat this value. store them in a const above and then use. it will reduce the code